services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -34,11 +34,36 @@ class ProjectService(projectRepository: ProjectRepository)(implicit executionCon
     projectRepository.addProject(project)
   }
 
-  def deactivateProjectById(projectId: ProjectId): Future[Option[Project]] =
-    for {
-      _ <- projectRepository.deactivateProjectById(projectId)
-      getProject <- projectRepository.getProjectById(projectId)
-    } yield getProject
+  def deactivateProjectById(projectId: ProjectId, userId: UserId): Future[Option[Project]] = {
+    val result = projectRepository.getProjectById(projectId)
+    result.flatMap {
+      case Some(project) if project.ownerId != userId => throw new ProjectAccessDeniedException
```

Comment:
Do not throw an exception, return `Future.failed` instead. Same for all other places in this PR.

Reply:
fixed

Reply:
Nothing has changed. Therefore, nothing is fixed.

Reply:
Sorry, I didn't push. Fixed now